### PR TITLE
Add review transaction method

### PR DESCRIPTION
**Pull request by neithanmo**
Add a new APDU instruction to review a transaction returning the hash upon buffer approval

### CODE REVIEW

**Review by emmanuelm41 — CHANGES_REQUESTED** [review]
Nat, can we check the way we do it now on dkgRound2 and implement this in the same way?

File: src/types.ts [hunk 1]
```diff
@@ -70,3 +71,6 @@ export interface ResponseDkgGetPublicPackage {
 export interface ResponseDkgBackupKeys {
   encryptedKeys: Buffer
 }
+export interface ResponseReviewTransaction {
+  hash?: Buffer
```
Comment (emmanuelm41) [inline src/types.ts]: No need to make this optional, as it always will have a value

File: src/index.ts [hunk 2]
```diff
@@ -63,6 +64,7 @@ export default class IronfishApp extends GenericApp {
         DKG_BACKUP_KEYS: 0x19,
         DKG_RESTORE_KEYS: 0x1a,
         GET_RESULT: 0x1b,
+        REVIEW_TX: 0x1c,
```
Comment (emmanuelm41) [inline src/index.ts:67]: I think we have to add this in types.ts too

File: src/deserialize.ts [hunk 3]
```diff
@@ -35,3 +37,14 @@ export const deserializeDkgRound2 = (data?: Buffer) => {
     publicPackage,
   }
 }
+
+export const deserializeReviewTx = (data?: Buffer) => {
```
Comment (emmanuelm41) [inline src/deserialize.ts:41]: This is not a buffer, but a ResponsePayload. It implements the ByteStream class. Check how we read in other cases!

**Review by emmanuelm41 — CHANGES_REQUESTED** [review]
Just a few comments to address

File: src/types.ts (hunk 1 above)
Comment (neithanmo) [reply src/types.ts]: done!

File: src/deserialize.ts (hunk 3 above)
Comment (neithanmo) [reply src/deserialize.ts:41]: it is defined as a buffer in Round2?
`export const deserializeDkgRound2 = (data?: Buffer)`

Comment (emmanuelm41) [reply src/deserialize.ts:41]: It is correct! My bad!

File: src/index.ts (hunk 2 above)
Comment (emmanuelm41) [reply src/index.ts:67]: The definition is missed here 
https://github.com/Zondax/ledger-ironfish-js/blob/a9857eac0be6276d633c783231281e1e0e25de33/src/types.ts#L6C1-L18C2

**Review by emmanuelm41 — APPROVED** [review]
LGTM... there is one comment to resolve yet!

File: src/index.ts (hunk 2 above)
Comment (neithanmo) [reply src/index.ts:67]: did you refresh?
that change is listed in the files changed menu

**Pull request merged**